[bigassfan] Add support for standalone lights

mhilbush opened the pull request:
Signed-off-by: Mark Hilbush <mark@hilbush.com>

- Add support for standalone lights, including ON/OFF, brightness level, and color temperature. See community discussion [here](https://community.openhab.org/t/haiku-light-with-bigassfan-binding/41226/3).

- Refactor config parameters into separate config.xml.

- Fix issues from code analysis

martinvw commented on addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java:
```diff
@@ -483,6 +461,17 @@ private boolean isLightPresent() {
         return false;
     }
 
+    private static final StringType LIGHT_COLOR = new StringType("COLOR");
```
I seem to have missed this the other time the normal order (element of least surprise) is that a class first contains all constants, then all fields, then constructor(s) and then the normal methods om some sensible order.

mhilbush replied:
Ok

martinvw commented on addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java:
```diff
@@ -483,6 +461,17 @@ private boolean isLightPresent() {
+    private static final StringType LIGHT_COLOR = new StringType("COLOR");
+
+    private boolean isLightColor() {
+        if (fanStateMap.containsKey(CHANNEL_LIGHT_COLOR)) {
```
This is the same as:

```java
return fanStateMap.containsKey(CHANNEL_LIGHT_COLOR) && LIGHT_COLOR.equals(fanStateMap.get(CHANNEL_LIGHT_COLOR))
```

Note that I also inverted the equals to make it more null-safe

mhilbush replied:
> Note that I also inverted the equals to make it more null-safe

Good catch. Thanks.

martinvw commented on addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java:
```diff
@@ -565,6 +554,8 @@ private ThingStatusDetail getDetail() {
         private final long FAN_LISTENER_DELAY = 2L;
```
Can the FanListener class get its own file?

mhilbush replied:
I was going to do that in a separate PR. Ok?

martinvw commented on addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/utils/BigAssFanConverter.java:
```diff
@@ -0,0 +1,71 @@
+    /*
+     * Convert from hue range (2200-5000) to dimmer range (0-100).
+     */
+    private static final double HUE_CONVERSION_FACTOR = 28.0;
```
Please put the constants together on top of the class

mhilbush replied:
Ok.

mhilbush commented:
@martinvw Thanks for the feedback. I addressed all of your comments except one, which I would prefer to do in a separate PR rather than this one. Hope that's ok with you.

martinvw commented:
Thank you too.
